src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -15788,30 +15788,33 @@ namespace SurfaceGeometry {
-                                                surfaceTmp.Name));
-                        int Np1 = n + 1;
-                        if (Np1 > NSides) {
-                            Np1 -= NSides;
-                        }
-                        int Np2 = n + 2;
-                        if (Np2 > NSides) {
-                            Np2 -= NSides;
-                        }
-                        ShowContinueError(state, format("...vertex {} to vertex {} to vertex {}", n, Np1, Np2));
-                        ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", n, X(n), Y(n), Z(n)));
-                        ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", Np1, X(n + 1), Y(n + 1), Z(n + 1)));
-                        ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", Np2, X(n + 2), Y(n + 2), Z(n + 2)));
-                        // ShowContinueError(state, format("...theta angle=[{:.6R}]", Theta));
-                        // ShowContinueError(state, format("...last theta angle=[{:.6R}]", LastTheta));
-                    }
+                if (state.dataGlobal->DisplayExtraWarnings && surfaceTmp.ExtSolar &&
+                    (state.dataHeatBal->SolarDistribution != DataHeatBalance::Shadowing::Minimal) &&
+                    // Warn only once
+                    surfaceTmp.IsConvex) {
+                    ShowWarningError(state,
+                                     format("CheckConvexity: Zone=\"{}\", Surface=\"{}\" is non-convex.",
+                                            state.dataHeatBal->Zone(surfaceTmp.Zone).Name,
+                                            surfaceTmp.Name));
+                    int Np1 = n + 1;
+                    if (Np1 > NSides) {
+                        Np1 -= NSides;
+                    }
+                    int Np2 = n + 2;
+                    if (Np2 > NSides) {
+                        Np2 -= NSides;
+                    }
+                    ShowContinueError(state, format("...vertex {} to vertex {} to vertex {}", n, Np1, Np2));
+                    ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", n, X(n), Y(n), Z(n)));
+                    ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", Np1, X(n + 1), Y(n + 1), Z(n + 1)));
+                    ShowContinueError(state, format("...vertex {}=[{:.2R},{:.2R},{:.2R}]", Np2, X(n + 2), Y(n + 2), Z(n + 2)));
+                    // ShowContinueError(state, format("...theta angle=[{:.6R}]", Theta));
+                    // ShowContinueError(state, format("...last theta angle=[{:.6R}]", LastTheta));
                 }
                 surfaceTmp.IsConvex = false;
-                break;
+                // #10103 - We do not want to break early, because we do want to consistently remove colinear vertices
+                // to avoid potential vertex size mismatch fatal errors
+                // break;
```

Comment:
And do NOT break, so we ensure we do drop all colinear vertices and do not end up with a vertex size mistmatch.